sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -489,8 +490,16 @@ public List<BoundedSource<Document>> split(
             // maxChunkSize is the Mongo partition size in MB
             LOG.debug("Splitting in chunk of {} MB", desiredBundleSizeBytes / 1024 / 1024);
             splitVectorCommand.append("maxChunkSize", desiredBundleSizeBytes / 1024 / 1024);
-            Document splitVectorCommandResult = mongoDatabase.runCommand(splitVectorCommand);
-            splitKeys = (List<Document>) splitVectorCommandResult.get("splitKeys");
+            try {
```

Comment:
Can we add a test case for this?
Also, can we add documentation to the log, so if a user sees this, they have context for why their command isn't supported?

Finally, should we universally not-fail on a 115 here, or are there potential cases where a "command not supported" exception would still be an exception?

Reply:
> Can we add a test case for this?
How would it be? cosmosdb is a SAAS service, what do you advise?

> Also, can we add documentation to the log, so if a user sees this, they have context for why their command isn't supported?
mmm... in what log? in the CHANGES.md?


> Finally, should we universally not-fail on a 115 here, or are there potential cases where a "command not supported" exception would still be an exception?
This is product of the `splitVector` command into the try catch, no other command is executed here.

Reply:
there should be a test case where we mock in the 115 exception, and verify the return of just the singleton list.

In the log where we say the command isn't supported, we should say why it isn't supported.

Reply:
> there should be a test case where we mock in the 115 exception, and verify the return of just the singleton list.
I already added it, you can check it

> In the log where we say the command isn't supported, we should say why it isn't supported.
I can't find a way to simulate this, can you help me?

Reply:
They unit test only work with one cosmosdb embebed versión.
This case just happen con cosmos or some implementation not standard, only for the command `splitVector` @johnjcasey

Reply:
In principle, this should be written such that we can mock out the database, so we can throw an exception when we want. It looks like the existing tests use an embedded instance of mongo. Is there any way we can tell that instance to throw an exception?

Reply:
@johnjcasey, please, Can I consider this conversation resolved?

Reply:
@johnjcasey, please, for the case where `splitKeys.size() < 1` also place `return Collections.singletonList(this);`

Reply:
@johnjcasey some other suggestion?